renderdoc/driver/vulkan/vk_core.h

```diff
@@ -691,6 +691,22 @@ class WrappedVulkan : public IFrameCapturer
 
   struct BakedCmdBufferInfo
   {
+    using PushDescriptorIDArray = rdcfixedarray<ResourceId, 64>;
+
+    class PushDescriptorBindPointArray : public rdcfixedarray<PushDescriptorIDArray, 3>
+    {
+    public:
+      using rdcfixedarray<PushDescriptorIDArray, 3>::operator[];
+      PushDescriptorIDArray &operator[](VkPipelineBindPoint bp)
+      {
+        return (*this)[bp == VK_PIPELINE_BIND_POINT_RAY_TRACING_KHR ? 2 : bp];
+      }
+      const PushDescriptorIDArray &operator[](VkPipelineBindPoint bp) const
+      {
+        return (*this)[bp == VK_PIPELINE_BIND_POINT_RAY_TRACING_KHR ? 2 : bp];
+      }
+    };
```

Comment:
This array is only accessed in a few places and only initialised once (per command buffer). I'd prefer you make the smaller change of just changing the array to `[3][64]` and then having a `ResourceId GetPushDescriptorID(VkPipelineBindPoint bindpoint, uint32_t set)` helper that does this lookup, rather than adding this extra container struct.

That should also simplify/eliminate the need for a number of the other changes.